Nautilus/Patchers/VehicleUpgradesPatcher.cs

```diff
@@ -133,59 +171,111 @@ private static void LazyInitialize(Vehicle __instance)
+            return;
 
-            float quickSlotCharge = __instance.quickSlotCharge[slotID];
-            float chargeScalar = __instance.GetSlotCharge(slotID);
+        quickSlotCharge = __instance.quickSlotCharge[slotID];
+        chargeScalar = __instance.GetSlotCharge(slotID);
 
-            moduleGadget.delegateOnUsed?.Invoke(__instance, slotID, quickSlotCharge, chargeScalar);
+        moduleGadget.delegateOnUsed?.Invoke(__instance, slotID, quickSlotCharge, chargeScalar);
 
-            var cooldown = 0f;
-            if (moduleGadget.Cooldown > 0f)
-                cooldown = (float) moduleGadget.Cooldown;
+        cooldown = 0f;
+        if (moduleGadget.Cooldown > 0f)
+            cooldown = (float) moduleGadget.Cooldown;
 
-            __instance.quickSlotTimeUsed[slotID] = Time.time;
-            __instance.quickSlotCooldown[slotID] = cooldown;
+        __instance.quickSlotTimeUsed[slotID] = Time.time;
+        __instance.quickSlotCooldown[slotID] = cooldown;
 
-            return false;
-        }
+        return;
+    }
+
+    /*
+     * The transpiler that doesn't work, it is feared by everyone, even the most advanced developers...
+    [HarmonyDebug]
+    [HarmonyTranspiler]
+    [HarmonyPatch(typeof(Vehicle), nameof(Vehicle.SlotKeyDown))]
+    private static IEnumerable<CodeInstruction> VehicleSlotKeyDown(IEnumerable<CodeInstruction> instructions)
+    {
+        var matcher = new CodeMatcher(instructions);
+
+
+        // V2
+        matcher.MatchForward(false,
+                new CodeMatch(OpCodes.Ldloc_1),
+                new CodeMatch(OpCodes.Ldc_I4_2),
+                new CodeMatch(i => i.opcode == OpCodes.Bne_Un_S));
+        var jumpTo = (Label) matcher.Operand;
+        matcher.RemoveInstructions(3)
+            .Insert(
+                new CodeInstruction(OpCodes.Ldloc_1),
+                new CodeInstruction(OpCodes.Ldc_I4_2),
+                new CodeInstruction(OpCodes.Ceq),
+                new CodeInstruction
+                new CodeInstruction(OpCodes.Ldloc_1),
+                new CodeInstruction(OpCodes.Ldc_I4_5),
+                new CodeInstruction(OpCodes.Bne_Un_S, jumpTo)
+            );
+
+        // V1
+        *
+        matcher.MatchForward(true,
+                new CodeMatch(OpCodes.Ldloc_1),
+                new CodeMatch(OpCodes.Ldc_I4_2),
+                new CodeMatch(OpCodes.Bne_Un_S)).CreateLabel(out Label jumpTo)
+            .Advance(2)
+            .Insert(
+                new CodeInstruction(OpCodes.Beq_S, ""),  // This is replacing the if(quickSlotType == QuickSlotTypes.Instant) with
+                new CodeInstruction(OpCodes.Ldloc_1),  // if(quickSlotType == QuickSlotTypes.Instant || quickSlotType == QuickSlotTypes.Chargeable)
+                new CodeInstruction(OpCodes.Ldc_I4_5)  // This is kinda advanced OpCodes operation, please do not touch with it unless you know what you're doing.
+            );
+        *
+
+        return matcher.InstructionEnumeration();
     }
+    */
```

Comment:
Not very clean to leave commented out code unless it is almost done and you plan on picking it up very soon

Reply:
ooh yeah my bad i'll remove that